### ISSUE TITLE
Getting error CRYPTO_STATIC_MUTEX_lock_write when using sdk with other http lib cpr (using curl + openssl)

### ISSUE THREAD

**Issue by mayurpatelgec** [aws/aws-iot-device-sdk-cpp-v2#301]
Confirm by changing [ ] to [x] below:
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-iot-device-sdk-cpp-v2/issues/) and didn't find any solution

**Known Issue**
- [x] I'm using ATS data type endpoint: the endpoint should look like `<prefix>-ats.iot.<region>.amazonaws.com`

**Platform/OS/Device**
What are you running the sdk on?
ARCH: Amd64
OS: Ubuntu:20.04 
AWS-SDK : 1.13.0 (latest at the moment)

**Describe the question**
Hi All,

> I dont know if its the right place to post the que here but its due to a lib conflict thats why m posting it here / or can help someone in future and to find some help from community. 

I am using this AWS-SDK since a year so m lil familiar with it now. There is also no issues with compilation and all. Its All working fine. Now, I want to integrate one HTTP client based on curl called project `CPR` to perform other HTTP/HTTPS related tasks. 

**Info about this lib cpr**
- lib README: https://github.com/whoshuu/cpr
- c++ wrapper around curl 
- uses openssl for https

**How I did it (example code)**
- I used our basic-pub-sub mqtt example (https://github.com/aws/aws-iot-device-sdk-cpp-v2/tree/main/samples/mqtt/basic_pub_sub) to start.
- added this small code to test it.
 ```
#include <cpr/cpr.h>
****
****
int test_http {
    cpr::Response r = cpr::Get(cpr::Url{"https://api.github.com/repos/whoshuu/cpr/contributors"},
                      cpr::Authentication{"user", "pass"},
                      cpr::Parameters{{"anon", "true"}, {"key", "value"}});
    r.status_code;                  // 200
    r.header["content-type"];       // application/json; charset=utf-8
    r.text;                         // JSON text string
}
```
- compiled it and linked the lib as described on their website README and it all went smoothly.

**Problem**
- when i run the code without any AWS-SDK code just only CPR lib demo then the code runs fine and work as expected.
- When I run this our above described code, then it throws a `CRYPTO_STATIC_MUTEX_lock_write` and then it crashes.
- screenshot of gdb trace:
![image](https://user-images.githubusercontent.com/26649420/126276557-bc5231f6-6efd-4c54-b65f-ee71757bdade.png)

**Hints(why this can happen)**
I am not an expert in libcrypto but still m putting my assumptions here
- can be due to multi threaded/mutex problems with curl (because curl does not support it and some threads can be already in use by AWS-SDK)
- conflicts between aws-lc (libcrypto) and libcrypto from cpr lib

**If anyone has hints about solving this problem then please let me know.
Thank you in advance and wish you all a great working day.**

**Comment by bretambrose** [comment]
Not knowing how you're building (shared vs static linking), but it's almost certainly a conflict between aws-lc and openssl's libcrypto.   You might try building aws-crt-cpp with this option enabled: https://github.com/awslabs/aws-crt-cpp/blob/main/CMakeLists.txt#L9

**Comment by mayurpatelgec** [comment]
Hi @bretambrose,

Thanx for the fast response. :)
Are you sure that its due to conflict and not any other issue like multi-threading etc. ?

**Comment by mayurpatelgec** [comment]
if I compile SDK with `-DUSE_OPENSSL=ON`
then it shows below error
![image](https://user-images.githubusercontent.com/26649420/126446474-e836d05e-fbbe-4e73-b365-fd47e5c1f48c.png)

I ve found a similar issue where you hv already replied @bretambrose : https://github.com/aws/aws-iot-device-sdk-cpp-v2/issues/258
its already solved but why m getting still the same error?

**Comment by bretambrose** [comment]
Well in their case, they were intending to use aws-lc but the cmake exports weren't being done properly and they instead ended up linking against system libcrypto.  I haven't actually tried using openssl since the aws-lc switch; I'll experiment with it later today and see if I get similar behavior.

**Comment by mayurpatelgec** [comment]
Hey, I found my mistake.

I compiled it in fresh docker and it was successful with 1.1.1f openssl. 

I was trying then to figure it out why was that failing on my host ubuntu and suddenly i saw openssl header with AWS flags and realized that somehow accidentally i installed AWS version of openssl in `usr/local/include` which was really bad (I remember that in older versions of sdk you had to provide `sudo` rights to install crypto OR provide install prefix to install it in DIR).

not now corrected it and got back default/system openssl back. Now will compile it and will try again with my main issue mentioned above.
